packages/api-bindings/src/apollo/publicationTypePolicy.ts

```diff
@@ -66,38 +68,26 @@ const collectPolicy = <T extends Post | Mirror | Comment>(
 ) => {
   if (existing) return existing;
 
-  const profile = readField('profile') as PublicationFragment['profile'];
-  const collectModule = readField('collectModule') as PublicationFragment['__collectModule'];
-  const publicationStats = readField('stats') as PublicationFragment['stats'];
+  const profile = readField('profile') as ProfileFragment;
```

Comment:
nit: we could at some point fix the typings to be fully typed (use `FieldFunctionOptions` under the hood so the `readField` returns correct types)

Reply:
I gave it a couple of tries in the last weeks. It's kinda tricky, simply because `TypePolicy` and `FieldPolicy` looks at the original GQL node and edges definitions. While the data retrieved from the cache are fragments, which in the best case scenarios are a subset of the nodes, but they could also be a superset of the node definitions if they include aliased fields.